src/Persistence/PersistentObjectFactory.php

```diff
@@ -362,36 +361,26 @@ protected function normalizeCollection(string $field, FactoryCollection $collect
     }
 
     /**
+     * This method will try to find entities in database if they are detached.
+     *
      * @internal
      */
     protected function normalizeObject(object $object): object
     {
-        $reflectionClass = new \ReflectionClass($object::class);
-
-        if ($reflectionClass->isFinal()) {
-            return $object;
-        }
-
-        // readonly classes exist since php 8.2 and proxyHelper supports them since 8.3
-        if (80200 <= \PHP_VERSION_ID && \PHP_VERSION_ID < 80300 && $reflectionClass->isReadonly()) {
-            return $object;
-        }
```

Comment:
I polished a bit this method...

this was not needed anymore since [this PR](https://github.com/zenstruck/foundry/pull/773), where we don't use proxy stuff anymore to refresh the entity